tests/lib/Strategy/FocusMode/ContentStructureRedirectStrategyTest.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Tests\AdminUi\Strategy\FocusMode;
+
+use Ibexa\AdminUi\Strategy\FocusMode\ContentStructureRedirectStrategy;
+use Ibexa\Contracts\AdminUi\FocusMode\RedirectStrategyInterface;
+use Ibexa\Contracts\Core\Repository\LocationService;
+use Ibexa\Contracts\Core\Repository\Values\Content\ContentInfo;
+use Ibexa\Contracts\Core\SiteAccess\ConfigResolverInterface;
+use Ibexa\Core\Repository\Values\Content\Location;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Routing\RouterInterface;
+
+final class ContentStructureRedirectStrategyTest extends TestCase
+{
+    private RedirectStrategyInterface $strategy;
+
+    protected function setUp(): void
+    {
+        $this->strategy = new ContentStructureRedirectStrategy(
+            $this->createConfigResolverMock(),
+            $this->createLocationServiceMock(),
+            $this->createRouterMock()
+        );
+    }
+
+    /**
+     * @dataProvider dataProviderForTestSupports
+     *
+     * @param array<string, string> $pathData
+     */
+    public function testSupports(array $pathData, bool $doesSupport): void
+    {
+        self::assertSame($doesSupport, $this->strategy->supports($pathData));
+    }
+
+    /**
+     * @return iterable<array{array<string, string>, bool}>
+     */
+    public function dataProviderForTestSupports(): iterable
+    {
+        yield [['_route' => 'foo.path'], false];
+
+        yield [['_route' => 'ibexa.content.view'], false];
+
+        yield [['_route' => 'ibexa.section.list'], true];
+
+        yield [['_route' => 'ibexa.content_type_group.list'], true];
+
+        yield [['_route' => 'ibexa.object_state.groups.list'], true];
+
+        yield [['_route' => 'ibexa.content_type_group.view'], true];
```

Comment:
Nice to have: `yield 'foo.path' => [['_route' => 'foo.path'], false];` makes data sets named, which is more readable for GUI test runners in IDEs.